packages/@vuepress/markdown-loader/index.js

```diff
@@ -26,7 +30,10 @@ module.exports = function (src) {
   if (!markdown) {
     markdown = md()
   }
-
+  let config = loadConfig(path.resolve(sourceDir, '.vuepress'), false)
+  if (isFunction(config)) {
+    config = config(this)
+  }
```

Comment:
You shouldn't re-load user's config at the scope makrdown-loader, this loader should only focus on how to transpile the source markdown file to Vue component.

`extractHeaders` should be a option for this loader.

Reply:
Ill do some more digging, but I couldn't find any way to get to the `extractHeaders` option from within this file already.

Reply:
Are you saying it should be available already inside of the loader file if defined in the `.vuepress/config.js` file such as:
```
module.exports = ctx => ({
  markdown: {
    extractHeaders: ['h2','h3','h4','h5']
  }
})
```

if that is the case, I have looked at the `this` object and could not find it anywhere.